### PR TITLE
"Final Sample Cleanup" Echobot with Counter ASP.Net Core

### PR DESCRIPTION
**Pull request by daveta**
- [x] Is numbered and named correctly and following the spec
- [x] All files have been spell checked
- [x] All style cop/ linting issues have been addressed
- [x] Include all DCR changes
- [x] Explicit state save
- [x] Method name changes begin, end etc
- [x] Does not use mainDialog unless that file actually extends Dialog/ ComponentDialog
- [x] Compiles with latest packages
- [x] Runs per instructions in README
- [x] Add commented out code to use blob storage
- [x] Update to look for production/ local (bot configuration)
- [x] onTurnError at adapter
- [x] Include .env or appsettings.json or web.config with botFilePath and botFileSecret
- [x] Deployment scripts folder has been cleaned up and only includes msbot export files
- [x] README to include
   - [x] msbot clone steps
   - [x] Tools install/ uninstall
   - [x] Concise set of steps to run the sample
- [x] C# sample -- Include .bot file in the build output

### CODE REVIEW

File: csharp_dotnetcore/02.EchoBot-With-Counter/EchoBotWithCounter.csproj [hunk 1]
```diff
@@ -16,10 +16,12 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore" Version="2.1.3" />
     <PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.7" />
-    <PackageReference Include="Microsoft.Bot.Builder" Version="4.0.0.38762" />
-    <PackageReference Include="Microsoft.Bot.Builder.Integration.AspNet.Core" Version="4.0.0.38762" />
-    <PackageReference Include="Microsoft.Bot.Configuration" Version="4.0.0.38762" />
+    <PackageReference Include="Microsoft.Bot.Builder" Version="4.0.5-21" />
```
Comment (JasonSowers) [inline csharp_dotnetcore/02.EchoBot-With-Counter/EchoBotWithCounter.csproj]: 4.0.5

**Comment by JasonSowers** [comment]
@yochay updated, can we merge?

File: csharp_dotnetcore/02.echo-with-counter/BotConfiguration.bot [hunk 2]
```diff
@@ -2,6 +2,7 @@
     "name": "EchoBotWithCounter",
     "description": "Echo bot with counter sample.",
```
Comment (vishwacsena) [inline csharp_dotnetcore/02.echo-with-counter/BotConfiguration.bot]: {
    "name": "echobot-with-counter",
    "services": [
        {
            "type": "endpoint",
            "name": "development",
            "endpoint": "http://localhost:3978/api/messages",
            "appId": "",
            "appPassword": "",
            "id": "1"
        }
    ],
    "padlock": "",
    "version": "2.0"
}

File: csharp_dotnetcore/02.echo-with-counter/DeploymentScripts/MsbotClone/bot.recipe [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (vishwacsena) [inline csharp_dotnetcore/02.echo-with-counter/DeploymentScripts/MsbotClone/bot.recipe]: \botbuilder-samples\csharp_dotnetcore -> \botbuilder-samples\samples\csharp_dotnetcore

File: csharp_dotnetcore/02.echo-with-counter/README.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+# To try this sample
+- Clone the samples repository
+```bash
+git clone https://github.com/Microsoft/botbuilder-samples.git
+```
+- [Optional] Update the `appsettings.json` file under `BotBuilder-Samples\csharp_dotnetcore\02.EchoBot-With-Counter` with your botFileSecret.  For Asure Bot Service bots, you can find the botFileSecret under application settings.
+- 
+# Prerequisites
+## Visual Studio
+- Navigate to the samples folder (`BotBuilder-Samples\csharp_dotnetcore\02.EchoBot-With-Counter`) and open EchoBotWithCounter.csproj in Visual Studio.
```
Comment (vishwacsena) [inline csharp_dotnetcore/02.echo-with-counter/README.md]: botbuilder-samples\samples\

**Pull request merged**